ocaml/xapi/xapi_pvs_proxy.ml

```diff
@@ -22,8 +22,24 @@ let not_implemented x =
 let proxy_port_name bridge =
 	"pvs-" ^ bridge
 
+let get_accessible_cache_sr ~__context proxy =
+	let farm = Db.PVS_proxy.get_farm ~__context ~self:proxy in
```

Comment:
I believe this code implements the policy to find an SR for a host. It picks the first SR it finds that is a cache SR and accessible from the host. What happens if multiple SRs exist? Would it always pick the same to maximise the cache effect? Currently there are no guarantees for this. The selection also does not take shared vs non-shared SRs into account. I think it would be good to document the properties of the selection mechanism in a comment at least. The understanding of what happens is further complicated by the fact that `Db.PVS_farm.get_cache_storage` does not make any guarantees about the order in which it returns SRs - so what "first" means is not clear. I like to suggest to at least bind the predicate `(fun sr -> host_has_pbd ..)` to a name like `first`.